src/battle_ai_main.c

```diff
@@ -4399,7 +4399,13 @@ static u32 AI_CalcMoveEffectScore(u32 battlerAtk, u32 battlerDef, u32 move)
         break;
     case EFFECT_REVIVAL_BLESSING:
         if (GetFirstFaintedPartyIndex(battlerAtk) != PARTY_SIZE)
+        {
             ADJUST_SCORE(DECENT_EFFECT);
+            if (AI_DATA->shouldSwitch & battlerAtk) // Bad matchup
+                ADJUST_SCORE(DECENT_EFFECT);
+            if (AI_DATA->mostSuitableMonId != PARTY_SIZE) // Good mon to send in after
+                ADJUST_SCORE(DECENT_EFFECT);
```

Comment:
I believe the lowest possible fainting score is 5 so if all conditions apply this would go over it. would this be intentional? Maybe it would be enough if the checks in the if condition are just weak_effect? So they kinda act like supporting increases

Reply:
Yeah I like that better, definitely don't want to eclipse the fainting score. Good idea :)